src/components/MemoryBar.tsx

```diff
@@ -0,0 +1,32 @@
+import { ProgressBar } from '@/components/ProgressBar';
+import { formatInt } from '@/helpers/numbers';
+
+export function MemoryBar({
+  memoryUsed,
+  memoryAllocated,
+}: {
+  memoryUsed?: number | null | undefined;
+  memoryAllocated?: number | null | undefined;
+}) {
+  if (!memoryAllocated) {
+    return null;
+  }
+  const memoryUsedNum = memoryUsed || 0;
+  const mbRemaining = memoryAllocated - memoryUsedNum;
+  return (
+    <div className="margin-top-3" data-testid="memory-bar">
+      <p className="font-sans-3xs text-uppercase text-bold">Memory:</p>
+
+      <ProgressBar total={memoryAllocated} fill={memoryUsedNum} />
+
+      <div className="margin-top-1 display-flex flex-justify font-sans-3xs">
+        <div className="">
```

Comment:
[non-blocking comment] I don’t mind the commas! But/and I think it might be better to display memory in terabytes. Is that an option?

(If memory can appear in MB _or_ TB for organizations, it’d be 🆒 if we could conditionally switch between the two in the design. But I realize that might add complexity!)

Reply:
I like that we're trying to minimize dependencies, but perhaps something like https://github.com/75lb/byte-size would be helpful for this and other potential use cases in the dashboard?

Reply:
We could convert to TB, but my first thought was to compare with what the CF CLI tool returns, to keep parity for users who use both. That tool shows it in Gigabytes:
![Screenshot 2024-10-09 at 12 30 47 PM](https://github.com/user-attachments/assets/76cf152e-e95a-4aa8-bd86-7d32bdc81437)

Reply:
sorry, I think I was looking at an old version of the comment. GB it is!

Reply:
Totally agreed on parity — that’s a fantastic point. (And I’m sorry for the confusion: I edited my initial comment because I _meant_ gigabytes, but typed terabytes.)

Reply:
@hursey013 Agreed, bringing in a converter tool totally makes here, given how much we'll be doing this.

Reply:
Done!